File: nexus-maliopt/src/main/java/com/nexus/modules/textures/TextureModule.java
package com.nexus.modules.textures;

import com.maliopt.MaliOptMod;
import java.io.InputStream;

public class TextureModule {
    public static void load() {
        try {
            ClassLoader cl = TextureModule.class.getClassLoader();
            InputStream manifestStream = cl.getResourceAsStream("assets/maliopt/astc_manifest.json");
            if (manifestStream != null) {
                String manifest = new String(manifestStream.readAllBytes());
                String[] lines = manifest.split("\\n");
                for (String line : lines) {
                    if (line.contains("\"file\":")) {
                        String[] parts = line.split("\"");
                        if (parts.length >= 4) {
                            String astcPath = parts[3]; // ex: "textures_astc/block/stone.astc"
                            String resourcePath = "assets/maliopt/" + astcPath;
                            InputStream input = cl.getResourceAsStream(resourcePath);
                            if (input != null) {
                                byte[] bytes = input.readAllBytes();
                                String key = astcPath.replace(".astc", "");
                                ASTCTextureRegistry.put(key, bytes);
                            }
                        }
                    }
                }
                MaliOptMod.LOGGER.info("[TextureModule] {} texturas ASTC carregadas", ASTCTextureRegistry.count());
            }
        } catch (Exception e) {
            MaliOptMod.LOGGER.warn("[TextureModule] Erro ao carregar texturas ASTC: {}", e.getMessage());
        }
    }
}

File: nexus-textures/src/main/kotlin/com/nexus/modules/textures/TextureModule.kt
package com.nexus.modules.textures

import org.slf4j.LoggerFactory

object TextureModule {
    private val logger = LoggerFactory.getLogger("nexus-textures")

    fun load() {
        try {
            val classLoader = TextureModule::class.java.classLoader
            val manifestStream = classLoader.getResourceAsStream("assets/maliopt/astc_manifest.json")
            if (manifestStream != null) {
                val manifest = String(manifestStream.readBytes())
                val lines = manifest.lines().filter { it.contains("\"file\":") }
                for (line in lines) {
                    val parts = line.split("\"")
                    if (parts.size >= 4) {
                        val astcPath = parts[3]  // ex: "textures_astc/block/stone.astc"
                        val resourcePath = "assets/maliopt/$astcPath"
                        val input = classLoader.getResourceAsStream(resourcePath)
                        if (input != null) {
                            val bytes = input.readBytes()
                            // A chave é o caminho relativo sem extensão
                            val key = astcPath.removeSuffix(".astc")
                            ASTCTextureRegistry.put(key, bytes)
                        }
                    }
                }
                logger.info("[TextureModule] {} texturas ASTC carregadas do mod", ASTCTextureRegistry.count())
            }
        } catch (e: Exception) {
            logger.warn("[TextureModule] Erro ao carregar texturas ASTC: {}", e.message)
        }
    }
}
